src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -1859,6 +1859,10 @@ public void calculate(CompilationUnit sourceCompilationUnit, List<CtType<?>> typ
 		}
 
 		Set<CtReference> imports = new HashSet<>();
+		if (sourceCompilationUnit != null) {
+			imports.addAll(sourceCompilationUnit.getImports());
```

Comment:
Will it work also in cases when AST is refactored and some classes will be removed? Then it will probably produce unused imports, will not?